src/hooks/useDetectOs.ts

```diff
@@ -4,11 +4,28 @@ import downloadUrlByOS from '../util/downloadUrlByOS';
 
 export const useDetectOs = () => {
   const [userOS, setUserOS] = useState<UserOS>(UserOS.UNKNOWN);
+  const [bitness, setBitness] = useState<string>('');
 
-  useEffect(() => setUserOS(detectOS), []);
+  useEffect(() => {
+    setUserOS(detectOS);
+
+    // This is necessary to detect Windows 11 on Edge.
+    // [MDN](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/getHighEntropyValues)
+    // [MSFT](https://learn.microsoft.com/en-us/microsoft-edge/web-platform/how-to-detect-win11)
+    // @ts-expect-error no types for "userAgentData" because this API is experimental
+    navigator?.userAgentData
```

Comment:
Do we really need optional chaining after `navigator` because navigator will never be undefined on client 🤔 
```suggestion
    navigator.userAgentData
```

Reply:
And the `useEffect` ensures it is client-side only. So indeed, no chaining needed here?

Reply:
Fixed in [f647291](https://github.com/nodejs/nodejs.dev/pull/3163/commits/f6472917dc884ec3c659652199bb1bb829feed2f)